File: custom-addons/morenitapp/models/localidad.py
from odoo import models, fields, api
from odoo.exceptions import ValidationError

# Modelo Localidad
# Guarda localidades/poblaciones

class Localidad(models.Model):

    _name = 'morenitapp.localidad'
    _description = 'Localidad de la Hermandad'

    # Campo representativo del registro
    _rec_name = 'nombreLocalidad'

    # Nombre de la localidad
    nombreLocalidad = fields.Char(
        required=True
    )

    # Relación con provincia
    codProvincia_id = fields.Many2one(
        'morenitapp.provincia',
        string="Provincia",
        required=True
    )

File: mobile_flutter/morenitapp/lib/features/panel-gestion/ubicaciones/domain/entities/localidad.dart
class Localidad {
  final int id;
  final String nombreLocalidad;
  final int codProvinciaId;

  Localidad({
    this.id = 0,
    required this.nombreLocalidad,
    required this.codProvinciaId,
  });

  factory Localidad.fromJson(Map<String, dynamic> json) {
  return Localidad(
    id: json['id'] ?? 0,
    nombreLocalidad: json['nombreLocalidad'] is String ? json['nombreLocalidad'] : '',
    codProvinciaId: json['codProvincia_id'] is List
        ? json['codProvincia_id'][0]
        : (json['codProvincia_id'] ?? 0),
  );
}

  Map<String, dynamic> toJson() => {
        'nombreLocalidad': nombreLocalidad,
        'codProvincia_id': codProvinciaId,
      };

  Localidad copyWith({
    int? id,
    String? nombreLocalidad,
    int? codProvinciaId,
  }) {
    return Localidad(
      id: id ?? this.id,
      nombreLocalidad: nombreLocalidad ?? this.nombreLocalidad,
      codProvinciaId: codProvinciaId ?? this.codProvinciaId,
    );
  }
}
